packages/medusa/src/services/shipping-option.ts

```diff
@@ -236,13 +315,13 @@ class ShippingOptionService extends BaseService {
       )
 
       let methodPrice
-      if ("price" in config) {
+      if ("price" in config && config.price !== undefined) {
         methodPrice = config.price
       } else {
         methodPrice = await this.getPrice_(option, validatedData, config.cart)
       }
 
-      const toCreate = {
+      const toCreate: CreateShippingMethod = {
```

Comment:
Since we are passing `toCreate` to `repository.create` does it make sense to have an intermediate type here? 

My thinking is that we could get away with something like `DeepPartial<ShippingMethod>` as this is what the `create` method is expecting. This way we avoid potential inconsistencies between `CreateShippingMethod` and `DeepPartial<ShippingMethod>`. 

I might have missed something so please let me know if the thinking here is not correct :)

Reply:
Yes, i very much agree with this, `DeepPartial` is the way to go, and we should strive to include these database entities where we can

Reply:
I think this is definitely the right approach